apis/scheduling/v1alpha1/pod_migration_job_types.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type PodMigrationJobSpec struct {
+	// Paused indicates whether the PodMigrationJob should to work or not.
+	// Default is false
+	// +optional
+	Paused bool `json:"paused,omitempty"`
+
+	// TimeoutInSeconds controls the PodMigrationJob timeout duration.
+	// +optional
+	TimeoutInSeconds *int32 `json:"timeoutInSeconds,omitempty"`
+
+	// Mode represents the operating mode of the Job
+	// Default is PodMigrationJobModeReservationFirst
+	// +optional
+	Mode PodMigrationJobMode `json:"mode,omitempty"`
+
+	// PodRef represents the Pod that be migrated
+	// +required
+	PodRef *corev1.ObjectReference `json:"podRef,omitempty"`
```

Comment:
to be required, or not to be. that is a question.

Reply:
fixed